bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/ScanAndCompareGarbageCollector.java

```diff
@@ -291,4 +288,55 @@ public void gc(GarbageCleaner garbageCleaner) {
         bkActiveledgers.removeAll(overReplicatedLedgers);
         return overReplicatedLedgers;
     }
+
+    private Set<Long> preCheckOverReplicatedLedgers(Set<Long> bkActiveLedgers) throws InterruptedException {
+        final Set<Long> candidateOverReplicatedLedgers = Sets.newHashSet();
+        final Semaphore semaphore = new Semaphore(MAX_CONCURRENT_ZK_REQUESTS);
+        final CountDownLatch latch = new CountDownLatch(bkActiveLedgers.size());
+
+        for (final Long ledgerId : bkActiveLedgers) {
+            try {
+                semaphore.acquire();
+                ledgerManager.readLedgerMetadata(ledgerId)
+                        .whenComplete((metadata, exception) -> {
+                            try {
+                                if (exception == null) {
+                                    if (isNotBookieIncludedInLedgerEnsembles(metadata)) {
+                                        candidateOverReplicatedLedgers.add(ledgerId);
+                                    }
+                                }
+                            } finally {
+                                semaphore.release();
+                                latch.countDown();
+                            }
+                        });
+            } catch (Throwable t) {
+                LOG.error("Exception when iterating through the ledgers to pre-check for over-replication", t);
+                latch.countDown();
+            }
+        }
+        latch.await();
+        LOG.info("Finished pre-check over-replicated ledgers. candidateOverReplicatedLedgersSize={}",
```

Comment:
Maybe it can be interesting to log how many ledgers are over replicated over the total number of active ledgers (bkActiveLedgers) 
something like

`Over-replicated ledgers pre-check count: {}/{},  candidateOverReplicatedLedgers.size(), bkActiveLedgers.size()`